.github/workflows/scan-dependencies.yml

```diff
@@ -0,0 +1,29 @@
+name: Scan Dependencies
+
+on:
+  push:
+    paths:
+      - '**/pom.xml'
+    branches:
+      - 'master'
+
+permissions:
+  contents: write
+
+jobs:
+  install:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Is all history needed for this check? Wouldn't the default of `fetch-depth: 1` suffice for this check?

Reply:
Based on https://github.com/actions/checkout
```yml
    # Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
```

Reply:
Oops, this was just copied from the build workflow - where we want fetch-depth 0 for sonar SCM analysis.